### PR TITLE
Configure the number of  image samples logged at each epoch and batch

### LINKED ISSUE #46: Configure sample image logging during training
For 2D models, logging one sample from each batch resulted in too few samples being logged for the validation set, since the number of FOVs for validation is relatively small with regard to typical batch sizes. However, changing the logging scheme to use the first N samples from only the first batch will log samples from only the first FOV in validation due to sliding window sampling. To support both use cases, I propose the following change to the logging interface:

Current:

```yaml
model:
  log_num_samples: 12
```

Change to:

```yaml
model:
  log_batches_per_epoch: 4
  log_samples_per_batch: 3
```

So for 2D training, these numbers can be (1, 12) so that even the validation epoch only has 1 batch, this will still log the same number of images.

### PR DESCRIPTION
**Pull request by ziw-liu**
Resolves #46.

### CODE REVIEW

**Review by edyoshikun — APPROVED** [review]
LGTM! thanks @ziw-liu
